src/main/java/com/clivelee/config/ArgumentProcessor.java

```diff
@@ -0,0 +1,24 @@
+package com.clivelee.config;
+
+import org.apache.http.util.TextUtils;
+
+import java.net.URI;
+
+public final class ArgumentProcessor {
+    public Configuration parseArguments(String[] args) {
+
+        if (args == null || args.length != 1) {
+            throw new UrlNotSpecifiedException();
+        }
+
+        String urlString = args[0];
+
+        URI uri = URI.create(urlString);
+
+        if (TextUtils.isEmpty(uri.getScheme())) {
+            urlString = "http://" + urlString;
```

Comment:
This would seem to limit the search shouldVisit function to http only, or if the domain name (incorrectly) specified https, it would limit it to https.  Can you fix?

Reply:
ah yes 👍